beacon-chain/rpc/beacon/validators.go

```diff
@@ -505,6 +509,48 @@ func (bs *Server) GetValidatorParticipation(
 	}, nil
 }
 
+// This appends non finalized block atts to state. To replay for a state, a node does not replay non canonical
+// nor finalized blocks. To count participation, this includes the attestations from the orphaned block to state.
+func (bs *Server) appendNonFinalizedBlockAttsToState(ctx context.Context, s *statetrie.BeaconState, e uint64) (*statetrie.BeaconState, error) {
+	start := helpers.StartSlot(e)
+	end := helpers.StartSlot(e + 1)
+	f := filters.NewFilter().SetStartSlot(start).SetEndSlot(end)
+	blks, err := bs.BeaconDB.Blocks(ctx, f)
+	if err != nil {
+		return nil, err
+	}
+	bRoots, err := bs.BeaconDB.BlockRoots(ctx, f)
```

Comment:
fyi , there is an issue with using this method. During the medalla incident there were times when the block roots did not match up to the blocks retrieved. #7028

Might be better to just hash them here.

Reply:
Will do